manifests/integrations/elasticsearch.pp

```diff
@@ -24,37 +24,26 @@
 #  }
 #
 class datadog_agent::integrations::elasticsearch(
-  $cluster_stats                       = false,
+  Boolean $cluster_stats               = false,
   Optional[String] $password           = undef,
-  $pending_task_stats                  = true,
-  $pshard_stats                        = false,
+  Boolean$pending_task_stats           = true,
+  Boolean $pshard_stats                = false,
   Optional[String] $ssl_cert           = undef,
   Optional[String] $ssl_key            = undef,
   Variant[Boolean, String] $ssl_verify = true,
-  $tags                                = [],
-  $url                                 = 'http://localhost:9200',
+  Array $tags                          = [],
+  String $url                          = 'http://localhost:9200',
```

Comment:
`$instances` below would be an `Array` or `Hash` (can't remember off the top of my head), but we should probably set the `Optional` type for it. I'd probably do the same for all  similar integrations. I'm a big fan of us setting types everywhere 👍

Reply:
I did only add types to those places were we had `validate_legacy`. I will add Optional[Array] to some `$instances` that were defaulted to `undef` but I won't add types everywhere as this might break setups where different types are passed.